### ISSUE TITLE
[DOCS-239] Specify eslint configuration file name

### ISSUE THREAD

**Issue by c-dinkel** [codacy/docs#584]
How do we tell codacy that our eslint config file is called `.eslintrc.cjs` ?

**Comment by c-dinkel** [comment]
![eslint-config-not-found](https://user-images.githubusercontent.com/20446628/113830765-39246e00-9787-11eb-8920-c5191d3ec365.png)

We cannot click the "Configuration file" button in order to tell codacy to use our custom configuration, even though our configuration file `.eslintrc.cjs` resides in our project root. How do we teach codacy our rules?

**Comment by prcr** [comment]
Hello @c-dinkel,

ESLint should [automatically detect the configuration file](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-file-formats) and then Codacy would let you select the option **Configuration file** on your Code patterns page.

Could you please tell us in which repository you're having this issue so we can have a look?

**Comment by c-dinkel** [comment]
We're having the issue with https://github.com/openscd/open-scd , the `eslint` config completely follows https://open-wc.org/ recommendations and comes from their default scaffold.

**Comment by prcr** [comment]
After looking deeper into this I realized that indeed Codacy doesn't detect `.eslintrc.cjs` configuration files at the moment (you can check the list of supported configuration files [on this table](https://docs.codacy.com/repositories-configure/code-patterns/#using-your-own-tool-configuration-files)). However, Codacy should detect all the configuration files that are supported by ESLint, so I added an item to our internal backlog to update this.

In the meantime, I suggest a quick workaround which is to use a `.eslintrc.yml` configuration file that extends your existing `.eslintrc.cjs`:

```
# Temporary workaround to use .eslintrc.cjs on Codacy
extends: .eslintrc.cjs
```

This way:

* ESLint detects the original `.eslintrc.cjs` file first according to [these priorities](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-file-formats) and doesn't complain about the ES module imports if you run the tool locally or on your own CI/CD workflows.
* The first configuration file that Codacy detects and passes along to ESLint is the `.yml` version, and ESLint uses the same settings that you already have on your `.cjs` configuration file.

Let us know if this works for you.

**Comment by c-dinkel** [comment]
Thanks for suggesting a workaround.

It does not work for us, since we lint according to`open-wc` standards, which means our (default scaffold) eslintrc imports their default config, which is not available in yml format.

**Comment by c-dinkel** [comment]
In the meantime I have closed the codacy account and removed the integration from the repo. We may consider the service again once this issue is resolved, so I'll continue to monitor here.

**Comment by prcr** [comment]
@c-dinkel, thank you for letting us know. :+1: We do intend to add support for all ESLint configuration file formats, so I'll leave this issue open and I'll report back when we release that update.

**Comment by prcr** [comment]
Hello @c-dinkel, I'm happy to let you know that Codacy now supports `.eslintrc.cjs` configuration files. Here's a test I did on a fork of the repository you shared above:

![image](https://user-images.githubusercontent.com/60105800/117677450-60b38f80-b1a6-11eb-80ed-51839ca42987.png)

However, when doing this test I realized that we don't support the plugin eslint-plugin-tsdoc yet. I suggest that you follow this other issue that I created for supporting the plugin if you're interested in giving Codacy another try in the future: https://github.com/codacy/codacy-eslint/issues/1423

In the meantime, I'm closing this issue. Thank you for taking the time to provide us with your feedback!
